Recipes/Recipes/ServiceLayer/NetworkService/NetworkService.swift

```diff
@@ -0,0 +1,148 @@
+    ///  - completionHandler: Замыкание, вызываемое после того, как будет завершен процесс запроса в сеть
+    func getDishRecipe(
+        categoryName: CategoryRecipeName,
+        searchSymbol: String?,
+        completionHandler: @escaping (Result<RecipeListDTO?, Error>) -> ()
+    )
+    /// Запрос рецепта по ссылке
+    /// - Parameters:
+    ///  - uri: Ссылка на  детальный рецепт
+    ///  - completionHandler: Замыкание, вызываемое после того, как будет завершен процесс запроса в сеть
+    func getDetailRecipe(
+        uri: String,
+        completionHandler: @escaping (Result<RecipeListDTO?, Error>) -> ()
+    )
+}
+
+/// Сервис для работы с сетью, запрашивает данные о рецептах
+final class NetworkService: NetworkServiceProtocol {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let scheme = "https"
+        static let host = "api.edamam.com"
+        static let categoryPath = "/api/recipes/v2"
+        static let detailPath = "/api/recipes/v2/by-uri"
+        static let typeQueryItem = "type"
+        static let typeValueQueryItem = "public"
+        static let appIdQueryItem = "app_id"
+        static let appIdValueQueryItem = "df3b8de1"
+        static let appKeyQueryItem = "app_key"
+        static let appKeyValueQueryItem = "eaf7a9b6dd78df72532332124fb0b972"
+        static let uriQueryItem = "uri"
+        static let dishTypeQueryItem = "dishType"
+        static let mainCourseTypeQueryItem = "Main course"
+        static let healthTypeQueryItem = "health"
+        static let qQueryItem = "q"
+    }
+
+    // MARK: - Public Methods
+
+    func getDishRecipe(
+        categoryName: CategoryRecipeName, searchSymbol: String? = nil,
+        completionHandler: @escaping (Result<RecipeListDTO?, Error>) -> ()
+    ) {
+        guard let url = createDishRecipeUrlComponent(categoryName, searchSymbol: searchSymbol).url else { return }
+
+        URLSession.shared.dataTask(with: URLRequest(url: url)) { data, _, error in
+            if let error = error {
+                completionHandler(.failure(error))
+                return
+            } else if let data = data {
+                do {
+                    let obj = try JSONDecoder().decode(RecipeListDTO.self, from: data)
+                    completionHandler(.success(obj))
+                } catch {
+                    completionHandler(.failure(error))
+                }
+            }
+        }.resume()
+    }
+
+    func getDetailRecipe(uri: String, completionHandler: @escaping (Result<RecipeListDTO?, Error>) -> ()) {
+        guard let url = createDetailRecipeUrlComponent(uri).url else { return }
+
+        URLSession.shared.dataTask(with: URLRequest(url: url)) { data, _, error in
+            if let error = error {
+                completionHandler(.failure(error))
+                return
+            } else if let data = data {
+                do {
+                    let obj = try JSONDecoder().decode(RecipeListDTO.self, from: data)
```

Comment:
```suggestion
                    let obj = try JSONDecoder().decode(RecipeListDTO.self, from: data)
```
тут делали 1 го рецепта не та модель у тебя